blueocean-core-js/src/js/fetch.js

```diff
@@ -170,16 +194,28 @@ export const Fetch = {
      * @param {Object} [options.fetchOptions] - Optional isomorphic-fetch options.
      * @returns JSON body.
      */
-    fetchJSON(url, { onSuccess, onError, fetchOptions } = {}) {
+    fetchJSON(url, { onSuccess, onError, fetchOptions, disableCapabilites } = {}) {
+        let fixedUrl = url;
+        if (urlconfig.getJenkinsRootURL() !== '' && !url.startsWith(urlconfig.getJenkinsRootURL())) {
+            fixedUrl = `${urlconfig.getJenkinsRootURL()}${url}`;
+        }
+        let future;
         if (!config.isJWTEnabled()) {
-            return FetchFunctions.rawFetchJSON(url, { onSuccess, onError, fetchOptions });
+            future = FetchFunctions.rawFetchJSON(fixedUrl, { onSuccess, onError, fetchOptions });
+        } else {
+            future = jwt.getToken()
+                .then(token => FetchFunctions.rawFetchJSON(fixedUrl, {
+                    onSuccess,
+                    onError,
+                    fetchOptions: FetchFunctions.jwtFetchOption(token, fetchOptions),
+                }));
         }
-        return jwt.getToken()
-            .then(token => FetchFunctions.rawFetchJSON(url, {
-                onSuccess,
-                onError,
-                fetchOptions: FetchFunctions.jwtFetchOption(token, fetchOptions),
-            }));
+
+        if (!disableCapabilites) {
+            return future.then(data => capabilityAugmenter.augmentCapabilities(utils.clone(data)));
```

Comment:
I don't believe it's necessary? While there might be multiple invocations of augmentCapabilities running around the same time (if there are multiple concurrent fetches for data) each invocation is going to operate on the data returned from each individual REST API call. It's probably not a big deal either way, although I'm not sure if `clone` could be slow for a large data set.